Source/Core/InputCommon/ControllerInterface/SDL/SDL.cpp

```diff
@@ -167,25 +172,6 @@ void PopulateDevices()
 Joystick::Joystick(SDL_Joystick* const joystick, const int sdl_index)
     : m_joystick(joystick), m_name(StripSpaces(GetJoystickName(sdl_index)))
 {
-// really bad HACKS:
```

Comment:
How is this hack gotten around now?

Reply:
I think the idea is that current SDL(2) has better suppot for xinput controllers, but I don't have one to verify that

Reply:
Ok.  I will try with my xbox360 controller

Reply:
We may want to keep this hack in?  At least when I picked the SDL device, it was not picking up my button presses in the controller mapping window.  This is with a xbox360 wired controller.

Reply:
can you check if re-adding that code makes it do what you expect?

Reply:
p.s. I think you should go into the "Configure Input" (right click on some button), then you should hopefully be able to see live readings 
![image](https://user-images.githubusercontent.com/113063/163662637-7ce54b03-a920-4e89-8f0c-8b1c26982ab5.png)

Reply:
Yes, I did go into live readings and pushed the button down.  None of the buttons got hit from what I could tell.

I'll try building this I guess.  It looks like this code just hides the SDL controller which would match what I see in master (I don't see a SDL controller in the drop down)

Reply:
Actually, oddly _none_ of the buttons are detected

Reply:
might be worth you (or someone with x360 controller) debugging.. :) i think it's supposed to work. maybe you could sanity check with some sdl2 test app or something

Reply:
maybe we just need some more `SDL_HINT_JOYSTICK_HIDAPI*` hints to sdl

Reply:
https://github.com/libsdl-org/SDL/issues/4970 this might also have something to do with it. this PR added `SDL_HapticOpenFromJoystick`, which is (maybe) the old way

Reply:
Well, I tried `SDL_JOYSTICK_HIDAPI_XBOX` but that didn't help.  I don't have a lot of interest in debugging this personally.

I do feel like this cleanup was a bit of a stretch given the original goal was just to support motion based controllers.  However, keeping this hack in also doesn't work because it seems the counts are now different :(.  Targeting just the name works to properly remove the SDL controller from the device list.

As an aside, I do wonder if we really need three different devices in the list for xbox controllers but agree someone should ideally bottom out why it doesn't work for SDL.

Reply:
Yea, after I've reviewed some of the SDL code, I can see the HIDAPI-related hints are pretty much all enabled by default (so the xbox one is already enabled).

if x360 controllers are still broken with SDL in this PR, i think it's good to keep the blocking code. can you indicate what would be good things to block on? still the name, `"xbox 360"`? what are the counts now?

Reply:
This works for my controller:

```
#ifdef _WIN32
  // checking the name is probably good (and hacky) enough
  // but I'll double check with the num of buttons/axes
  std::string lcasename = GetName();
  Common::ToLower(&lcasename);

  if ((std::string::npos != lcasename.find("xbox 360")) &&
      (11 == SDL_JoystickNumButtons(joystick)) && (6 == SDL_JoystickNumAxes(joystick)) &&
      (1 == SDL_JoystickNumHats(joystick)) && (0 == SDL_JoystickNumBalls(joystick)))
  {
    // this device won't be used
    return;
  }
#endif
```

no idea why there is an extra axes and button compared with before...

Reply:
I wonder if maybe this code was only on Linux before, and the xbox driver exposes different things there